src/validators/validators-functions.js

```diff
@@ -0,0 +1,56 @@
+import { isFunction, isPlainObject } from '../utils/type-inference';
+import { castArray } from '../utils/array';
+
+const getValidators = (validator) => isFunction(validator) ? validator() : validator;
+
+/**
+ * check if a value is valid or not
+ * @param  {object} validators -  a config object
+ * @param  {any} value
+ * @param  {key} string
+ * @returns boolean - using the validators to check if the value is a valid value or not
+ */
+export const isValueValid = (validators, value, key) => {
+  let isValid = false;
+  let invalidItem = null;
+
+  castArray(validators).forEach((validator) => {
+    const isItemValid = getValidators(validator).value(value);
+
+    isValid = isItemValid && !isValid ? isItemValid : isValid;
+    invalidItem = !isItemValid && !invalidItem ? validator : invalidItem;
+  });
+
+  if (invalidItem && !isValid) {
+    const validatorItem = getValidators(invalidItem);
+    console.error(`cloudinary video player: ${validatorItem.message(key)}`);
+  }
+
+  return isValid;
```

Comment:
I suggest to improve these:
1. explain what 'key' is and/or name it better
2. explain that this function returns true if at least one of the validators returns true and otherwise it throws, (which is weird because you'd expect it to return true if all validators return true, but i'm not familiar with the spec/need here)
3. split this function's concerns into functions that do a single job/concern OR write it in a declarative manner

Plus a question: why do you need an array of validator if you only care that one returns true? i don't see a test for an array of validators, are you sure its needed?

Reply:
I removed the need for a validators array. The only property that had an array was 'transformation' (can be both object and array) I do not think we should check it since it is a SDK configuration and not a 'video player' configuration.  I rename 'isObject' to 'isPlainObject' (true only for {} ) and change 'isObject' to be true for both object and array